source/common/quic/quic_filter_manager_connection_impl.cc

```diff
@@ -146,10 +146,6 @@ void QuicFilterManagerConnectionImpl::updateBytesBuffered(size_t old_buffered_by
 
 void QuicFilterManagerConnectionImpl::maybeApplyDelayClosePolicy() {
   if (!inDelayedClose()) {
-    if (close_type_during_initialize_.has_value()) {
-      close(close_type_during_initialize_.value());
-      close_type_during_initialize_ = absl::nullopt;
-    }
```

Comment:
Do you think there's any need for a feature flag to control this part of the change? Presumably you're confident that it's a safe change? Or should we guard it with the same flag?

Reply:
It's guarded by close_type_during_initialize_. If the current feature flag is on, close_type_during_initialize_ wont' have value here, as OnCanWrite() happens outside of ProcessUdpPacket().

Reply:
I don't think I understand. `maybeApplyDelayClosePolicy()` is called from a couple of different places. Before this PR, it's possible for `maybeApplyDelayClosePolicy()` to actually call `close()` if not in delayed close (on line 150). But with this PR, if we're not in delayed close then `maybeApplyDelayClosePolicy()` will not call `close()`.

But I guess you're saying that before this PR, `close_type_during_initialize_` will never have a value here. Do I have that right?

Reply:
maybeApplyDelayClosePolicy() is only called in OnCanWrite() which was called at the end of ProcessUdpPacket() and various other places before this PR. But in this PR, we check `close_type_during_initialize_` at the beginning of ProcessUdpPacket() and close connection there. So OnCanWrite() won't be called at all in ProcessUdpPacket(). This check only takes effect at processing the 1st packet, `close_type_during_initialize_` will always be nullptr afterwards.

Reply:
Ok, I see. Thanks for the explanation.